### ISSUE TITLE
Saved Instance State

### ISSUE THREAD

**Issue by l3uddz** [Yalantis/uCrop#117]
Hey guys, trying to diagnose an issue with my code (if there is indeed any).
After starting uCrop.start(....) I can see my instance state variables being saved, and when pressing the back button (nav bar) to exit uCrop it restores my activity's saved state correctly, also the same applies for uCrops "Save" / tick option at top right. 
However If I press the cross at top left of uCrop activity, my activity's saved state is empty?
Is this a bug with uCrop or something I must look into myself.

**Comment by shliama** [comment]
Hi @l3uddz ,
as you can see, app calls `onBackPressed();` method when you click "the cross at top left of uCrop activity", therefore it's the same flow as if you click "back button (nav bar)". 

Do you override `onRestoreInstanceState` method to get you state or get it inside the `onCreate` method?

**Comment by l3uddz** [comment]
Inside the onCreate method, have tried onRestore too... It seems savedInstanceState is null in onCreate when pressing the back button. Pressing navback back button has expected result.

**Comment by l3uddz** [comment]
Confirmed with the android debugger, savedInstanceState is null when pressing the X (Back button). WHen pressing the navigation bar's back button, the savedInstanceState is filled as expected.

**Comment by l3uddz** [comment]
Just wanted to say this is resolved. It seems unrelated to uCrop, which looking at the code it should be.
In my manifest, where I declared the uCrop activity, I had set the parentActivityName to the activity that would be starting uCrop. This seems to have caused the issue of saved instance state being set to null upon clicking the toolbar's back button. Now that I have removed parentActivityName from the uCrop activity in my manifest, everything is as it should be.
